inspirehep/records/api/literature.py

```diff
@@ -143,3 +169,141 @@ def _dump_for_es(self):
+        pid_value = normalize_doi(identifier)
+
+    else:
+        raise UnknownImportIdentifierError(identifier)
+
+    pid = PersistentIdentifier.query.filter_by(
+        pid_type=pid_type, pid_value=pid_value
+    ).one_or_none()
+
+    if pid:
+        raise ExistingArticleError(
+            f"Article {identifier} already in Inspire. UUID: {pid.object_uuid}"
+        )
+    importers = {"arxiv": import_arxiv, "doi": import_doi}
+    importer = importers.get(pid_type, UnknownImportIdentifierError)
+    article = importer(pid_value)
+
+    if not article:
+        raise ImportArticleError(f"No article found for {identifier}")
+    return article
+
+
+def import_arxiv(arxiv_id: str) -> dict:
+    """View for retrieving an article from arXiv.
+
+    This endpoint is designed to be queried by inspirehep during an article
+    submission, to auto-fill the submission form if the user provides an arXiv
+    identifier.
+
+    Args:
+        arxiv_id: the normalized arXiv id, e.g. `0804.2273`.
+
+    Returns:
+        dict: a json object with either the parsed article
+
+    Raises:
+        ImportConnectionError: if the request doesn't succeed.
+        ImportParsingError: if any error occurs during the response parsing.
+
+    """
+    url = ARXIV_URL.replace(PLACEHOLDER, arxiv_id)
+
+    try:
+        resp = requests.get(url=url)
+    except (ConnectionError, IOError) as e:
+        logger.log(
+            level=logging.ERROR, msg=f"Error importing {arxiv_id} at {url}. \n{e}"
+        )
+        raise ImportConnectionError(f"Cannot contact arXiv: {e}")
+
+    if resp.status_code >= 400:
+        raise ImportConnectionError(f"Cannot contact arXiv. Got response {resp}.")
+
+    if "Malformed identifier" in str(resp.text):
+        # arXiv will reply 200 for a non existing arXiv ID with a message error
+        return {}
+
+    try:
+        parser = ArxivParser(resp.text)
+        return parser.parse()
+    except Exception as e:
+        logging.log(level=logging.ERROR, msg=f"Error parsing arXiv {arxiv_id}. \n{e}")
+        raise ImportParsingError(
+            f"An error occurred while parsing article oai:arXiv.org:{arxiv_id}. Error: {e}."
+        )
+
+
+def import_doi(doi: str) -> dict:
+    """View for retrieving an article from CrossRef.
+
+    This endpoint is designed to be queried by inspirehep during an article
+    submission, to auto-fill the submission form if the user provides a DOI.
+
+    Args:
+        doi: a normalized DOI id, e.g. `10.1088/1361-6633/aa5514`. The
+        variable has type `path` in order to properly handle '/' in the param.
+
+    Returns:
+        dict: a json object with either the parsed article or the occurred
+        error.
+
+    Raises:
+        ImportConnectionError: if the request doesn't succeed.
+        ImportParsingError: if any error occurs during the response parsing.
+
+    """
+    doi = requests.utils.quote(doi, safe="")
+    url = CROSSREF_URL.replace(PLACEHOLDER, doi)
```

Comment:
I would go for `format` here.

Reply:
do you have a strong reason not to just inline url `string` with `format` here?

Reply:
I don't, what's yours? :)

Reply:
* `replace` seems unnecessary for a simple string contact
* it's easy to read when it's next to the place where it is used, since it's only used just here
* we use `format` everywhere else to create urls